security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java

```diff
@@ -107,8 +107,7 @@ public Object loadUserBySAML(SAMLCredential credential)
             if (attrName.equals(SAML_IDP_METADATA_ROLE_ATTR_NAME)) {
                 List<XMLObject> attributeValues = cAttribute.getAttributeValues();
                 if (!attributeValues.isEmpty()) {
-                    userRoles.add(new StringBuilder(APP_NAME).append(":").append(
-                        getAttributeValue(attributeValues.get(0))).toString());
+                    userRoles.add(getAttributeValue(attributeValues.get(0)));
```

Comment:
@khzhu as @fedde-s mentioned, he found a solution to the problem I mentioned in the first comment of this thread. Given that you are using only one portal, this change does not really affect your installation in any way. Based on what I see in the code, the `users` and `authorities` tables are also not used anymore in case your instance is configured to use Keycloak. I.e. your `SAMLUserDetailsServiceImpl.java` class parses all authorities from SAML token only, and as far as I checked this is the only part of the login workflow where authorities are filled. 

So, I think this change is correct and we should keep it. Could you please test it on your side with this new code and let us know if it works for you? Although I do not think it is the case here, there is always a chance we missed something, so a test on your side would be nice. Thanks!